src/internal.rs

```diff
@@ -14,7 +14,8 @@ pub(crate) mod timestamp_millis {
     ) -> Result<chrono::DateTime<chrono::Utc>, D::Error> {
         let millis = i64::deserialize(des)?;
         Ok(chrono::DateTime::<chrono::Utc>::from(std::time::UNIX_EPOCH)
-            + chrono::Duration::milliseconds(millis))
+            + chrono::TimeDelta::try_milliseconds(millis)
+                .ok_or(serde::de::Error::custom("Timestamp out of bounds!"))?)
```

Comment:
```suggestion
                .ok_or_else(|| serde::de::Error::custom("Timestamp out of bounds!"))?)
```

I'm not sure why clippy does not complain, but we should construct the error lazily since I think it requires a memory allocation.